shared/team-building/email-input/index.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react'
+import {debounce} from 'lodash-es'
+import * as Container from '../../util/container'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as TeamBuildingGen from '../../actions/team-building-gen'
+import {AllowedNamespace} from 'constants/types/team-building'
+import {requestIdleCallback} from '../../util/idle-callback'
+
+type EmailInputProps = {
+  namespace: AllowedNamespace
+}
+
+function checkValidEmail(str: string): boolean {
+  const emailRegex = /^(\S+@\S+\.\S+)$/
+  return str.length > 3 && emailRegex.test(str)
+}
+
+const EmailInput = ({namespace}: EmailInputProps) => {
+  const [isEmailValid, setEmailValidity] = React.useState(false)
+  const [emailString, setEmailString] = React.useState('')
+  const dispatch = Container.useDispatch()
+  const user = Container.useSelector(state => {
+    return state[namespace].teamBuilding.teamBuildingEmailResult
+  })
+  const isSearching = Container.useSelector(
+    state => state[namespace].teamBuilding.teamBuildingEmailIsSearching
+  )
+
+  const debouncedSearch = debounce(
+    (query: string) =>
+      requestIdleCallback(() => dispatch(TeamBuildingGen.createSearchEmailAddress({namespace, query}))),
+    200
+  )
+
+  const onChange = React.useCallback(
+    text => {
+      setEmailString(text)
+      const isNewInputValid = checkValidEmail(text)
+      if (isNewInputValid !== isEmailValid) {
+        setEmailValidity(isNewInputValid)
+      }
+      if (isNewInputValid) {
+        debouncedSearch(text)
+      }
+    },
+    [isEmailValid, debouncedSearch]
+  )
+
+  const onSubmit = React.useCallback(() => {
+    if (!user) {
+      throw new Error('User not found, cannot submit')
```

Comment:
We should never get into this state, but is there something more graceful we could do here than throw an error?